include/deal.II/lac/trilinos_tpetra_vector.h

```diff
@@ -659,45 +677,75 @@ namespace LinearAlgebra
        */
       /** @{ */
       /**
-       * Compress the underlying representation of the Trilinos object, i.e.
-       * flush the buffers of the vector object if it has any. This function is
-       * necessary after writing into a vector element-by-element and before
-       * anything else can be done on it.
+       * Communicate all changes made to ghost entries of the vector
+       * (i.e., entries included in the locally relevant index set but not in
+       * the locally owned index set). The VectorOperation operation specifies
+       * how to deal with entries that multiple processors modified. This is
+       * necessary whenever we are done modifying the vector data on multiple
+       * (MPI) processes via add() or set().
+       *
+       * This function compresses the data structures and allows the vector to
+       * be used in all other operations, like matrix-vector products. This is
+       * a collective operation, i.e., it needs to be run on all processors
+       * when used in %parallel.
+       *
+       * After compress() was called, each processor can only modify locally
+       * owned elements. To modify entries belonging to a different processor
+       * again, one must first call decompress().
        *
        * See
        * @ref GlossCompress "Compressing distributed objects"
        * for more information.
        */
       void
-      compress(const VectorOperation::values operation);
+      compress(const VectorOperation::values operation,
+               const IndexSet               &locally_owned = IndexSet());
```

Comment:
Is this basically the same doc text as for the other parallel vectors? It would be nice to have all of these functions say the same thing -- because ideally they would implement the same semantics.

Reply:
No, this is a different text. 
The compress works differently in the case of the Tpetra Vector.

But with the argument that most differences are in the background, so the user does not need to know them, we can insert here the default text.

Reply:
Yes, that would be my preference. What matters is not *how* it is done, but *what* it does from a user perspective. I hope that that is the same as for the other parallel vectors.

Reply:
I inserted the text from the (epetra-based) Trilinos Vector here.